packages/backend/src/app/webhooks/webhook-service.ts

```diff
@@ -1,72 +1,112 @@
 
 export const webhookService = {
     async callback({ flowId, payload }: CallbackParams): Promise<void> {
-        const flow = await flowRepo.findOneBy({ id: flowId });
-        if (flow === null || flowId === null || flowId === undefined) {
-            throw new ActivepiecesError({
-                code: ErrorCode.FLOW_NOT_FOUND,
-                params: {
-                    id: flowId,
-                },
-            });
-        }
-        triggerEventService.saveEvent({ flowId: flowId, payload, projectId: flow.projectId });
-        const collection = await collectionService.getOneOrThrow({ projectId: flow.projectId, id: flow.collectionId });
-        const instance = await instanceService.getByCollectionId({ projectId: flow.projectId, collectionId: collection.id });
-        if (instance === null) {
-            return;
-        }
-        const flowVersion = await flowVersionService.getOneOrThrow(
-            instance.flowIdToVersionId[flow.id]
-        );
+        const flow = await getFlowOrThrow(flowId);
+        const { projectId, collectionId } = flow;
+        const flowVersion = await getLatestFlowVersionOrThrow(flowId, projectId);
+
+        triggerEventService.saveEvent({
+            flowId,
+            payload,
+            projectId,
+        });
+
         const payloads: unknown[] = await triggerUtils.executeTrigger({
-            projectId: collection.projectId,
-            collectionId: collection.id,
-            flowVersion: flowVersion,
-            payload: payload
+            projectId,
+            collectionId,
+            flowVersion,
+            payload,
+            simulate: false,
         });
 
         const createFlowRuns = payloads.map((payload) =>
             flowRunService.start({
                 environment: RunEnvironment.PRODUCTION,
-                collectionId: collection.id,
+                collectionId,
                 flowVersionId: flowVersion.id,
                 payload,
             })
         );
 
         await Promise.all(createFlowRuns);
     },
-    async getWebhookPrefix(): Promise<string> {
+
+    async simulationCallback({ flowId, payload }: CallbackParams): Promise<void> {
+        const flow = await getFlowOrThrow(flowId);
+        const { projectId, collectionId } = flow;
+        const flowVersion = await getLatestFlowVersionOrThrow(flowId, projectId);
+
+        const events = await triggerUtils.executeTrigger({
+            projectId,
+            collectionId,
+            flowVersion,
+            payload,
+            simulate: true,
+        });
+
+        if (isEmpty(events)) {
+            return;
+        }
+
+        logger.debug(events, `[WebhookService#simulationCallback] events, flowId=${flowId}`);
+
+        const eventSaveJobs = events.map(event => triggerEventService.saveEvent({
+            flowId,
+            projectId,
+            payload: event,
+        }));
+
+        await Promise.all(eventSaveJobs);
+
+        await triggerUtils.disable({
```

Comment:
This bit optimistic, If it's called twice then it might cause an error.

We could surround it with TODO.

Reply:
done in 6698d64a